src/main/java/co/rsk/federate/watcher/FederationWatcherListener.java

```diff
@@ -0,0 +1,26 @@
+package co.rsk.federate.watcher;
+
+import co.rsk.peg.federation.Federation;
+
+/**
+ * A listener interface for receiving notifications about changes in federations.
+ * Implementers of this interface will be notified when the active or retiring federation changes.
+ */
+public interface FederationWatcherListener {
+
+    /**
+     * Invoked when the active federation changes.
+     *
+     * @param newFederation the new active federation after the change.
+     *                      This will never be {@code null}; the active federation is always present.
+     */
+    void onActiveFederationChange(Federation newFederation);
```

Comment:
```suggestion
    void onActiveFederationChange(Federation newActiveFederation);
```

maybe redundant, but at first reading it caused me some confusion. Just a suggestion, no big deal

Reply:
I think we can leave as is